src/lib/Libifl/ifl_util.c

```diff
@@ -145,30 +146,41 @@ random_srv_conn(svr_conn_t **svr_conns)
  * @retval < 0: could not find appropriate index
  */
 int
-get_job_location_hint(char *jobid)
+get_job_resv_location_hint(char *job_resv_id)
```

Comment:
I have refactored the function to work on either job id/resv id.  Also added a new parameter to specify the object type. 
This function also considers a job/resv id even if it is not qualified with PBS_SERVER name. Please look at my latest commit for more details.
To answer your other question, if we pass "0.pbspro" for single server configuration, the function returns early since we are not in Multi-Server mode. 
If we pass "0.pbspro" for multi-server configuration it returns -1 since the length of object id should be minimum >=2